subworkflows/local/parse_input.nf

```diff
@@ -1,167 +1,85 @@
-//
-// Check input samplesheet or folder and get read channels
-//
-
-// Function to get list of [ meta, [ fastq_1, fastq_2 ] ]
-def parse_samplesheet(LinkedHashMap row, single_end) {
-    //Check if samplesheet contains column sampleID  & forwardReads
-    if (row.sampleID == null || row.forwardReads == null) {
-        error("ERROR: Please check input samplesheet -> Column 'sampleID' and 'forwardReads' are required but not detected.")
-    }
-    //Check if samplesheet contains a column for reverse reads
-    if (row.reverseReads == null && !single_end) {
-        error("ERROR: Please check input samplesheet -> Column 'reverseReads' is missing. In case you do have only single ended reads, please specify '--single_end', '--pacbio', or '--iontorrent'.")
-    }
-    //Check if samplesheet contains a column run and empty fields
-    if (row.run != null && row.run == "") {
-        error("ERROR: Please check input samplesheet -> Column 'run' contains an empty field. Either remove column 'run' or fill each field with a value.")
-    }
-    //read meta info
-    def meta = [:]
-    meta.id           = row.sampleID
-    meta.single_end   = single_end.toBoolean()
-    meta.run          = row.run == null ? "1" : row.run
-    //read data info
-    def array = []
-    if (!file(row.forwardReads).exists()) {
-        error("ERROR: Please check input samplesheet -> Forward read FastQ file does not exist!\n${row.forwardReads}")
-    }
-    if (meta.single_end) {
-        array = [ meta, file(row.forwardReads) ]
-    } else {
-        if (!file(row.reverseReads).exists()) {
-            error("ERROR: Please check input samplesheet -> Reverse read FastQ file does not exist!\n${row.reverseReads}")
-        }
-        array = [ meta, [ file(row.forwardReads), file(row.reverseReads) ] ]
-    }
-    return array
-}
-
 workflow PARSE_INPUT {
     take:
-    input // file.tsv or folder
-    is_fasta_input
+    input // folder
```

Comment:
```suggestion
    input // sample sheet
```

Reply:
?

Reply:
Thanks for the review!
The `workflow PARSE_INPUT` accepted in the past indeed the sample sheet & folder input. But now the sample sheet is parsed via `--input` with the function [fromSamplesheet](https://github.com/nf-core/ampliseq/blob/a13662791add73e68650cd709fcf776618ab5346/workflows/ampliseq.nf#L244). Folder input is still processed with `workflow PARSE_INPUT`, see [here](https://github.com/nf-core/ampliseq/blob/a13662791add73e68650cd709fcf776618ab5346/workflows/ampliseq.nf#L254-L255), now via `--input_folder` instead of `--input`. So I think its fine as it is.